src/pybind/mgr/progress/module.py

```diff
@@ -735,5 +764,16 @@ def handle_command(self, _, cmd):
             return self._handle_clear()
         elif cmd['prefix'] == "progress json":
             return 0, json.dumps(self._json(), indent=4, sort_keys=True), ""
+        elif cmd['prefix'] == "progress on":
+            if self.enabled:
+                return 0, "", "progress already enabled!"
+            self.on()
+            return 0, "", "progress enabled"
+        elif cmd['prefix'] == "progress off":
+            if not self.enabled:
+                return 0, "", "progress already disabled!"
+            self.clear()
```

Comment:
order should be off(), then clear(), so that there's no chance of events being added between the calls (perhaps not an issue today, but it'd be more future-proof)